miio/gateway.py

```diff
@@ -646,71 +646,112 @@ class GatewayLight(GatewayDevice):
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command()
+    def night_light_status(self):
+        """Get status of the night light."""
+        # The night light controls the same LEDs as the rgb light, but has a seprate memory for brightness and color.
+        # Changing the rgb light does not effect the stored state of the night light, while changing the night light does effect the state of the rgb light.
+        # Therefore this command only gives the correct status of the LEDs if the last command was a night light command and not a rgb light command, otherwise it gives the stored values of the night light.
+        # Returns {"is_on": false, "brightness": 0, "rgb": (0, 0, 0)} when night light is off
+        state_int = self._gateway.send("get_night_light_rgb").pop()
+        brightness = int_to_brightness(state_int)
+        rgb = int_to_rgb(state_int)
+        if brightness > 0:
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command(
+        click.argument("brightness", type=int),
+        click.argument("rgb", type=(int, int, int)),
+    )
+    def set_rgb(self, brightness, rgb):
+        """Set gateway light using brightness and rgb tuple."""
+        brightness_and_color = brightness_and_color_to_int(brightness, rgb)
+        return self._gateway.send("set_rgb", [brightness_and_color])
+
+    @command(
+        click.argument("brightness", type=int),
+        click.argument("rgb", type=(int, int, int)),
+    )
+    def set_night_light(self, brightness, rgb):
+        """Set gateway night light using brightness and rgb tuple."""
+        brightness_and_color = brightness_and_color_to_int(brightness, rgb)
+        return self._gateway.send("set_night_light_rgb", [brightness_and_color])
+
+    @command(click.argument("brightness", type=int))
+    def set_rgb_brightness(self, brightness):
+        """Set gateway light brightness (0-100)."""
+        if 100 < brightness < 0:
+            raise Exception("Brightness must be between 0 and 100")
+        current_color = self.rgb_status()["rgb"]
+        return self.set_rgb(brightness, current_color)
+
+    @command(click.argument("brightness", type=int))
+    def set_night_light_brightness(self, brightness):
+        """Set night light brightness (0-100)."""
+        if 100 < brightness < 0:
+            raise Exception("Brightness must be between 0 and 100")
+        current_color = self.night_light_status()["rgb"]
+        return self.set_night_light(brightness, current_color)
 
     @command(click.argument("color_name", type=str))
-    def set_night_light_color(self, color_name):
-        """Set night light color using color name (red, green, etc)."""
+    def set_rgb_color(self, color_name):
+        """Set gateway light color using color name (red, green, etc)."""
```

Comment:
Is the list of colors static and are the valid values known? Maybe this should be enforced using an enum (or a list)?

Reply:
This is a static dict that is defined on line 26:
https://github.com/rytilahti/python-miio/blob/79e61fbba512a79b569fc5cf5d14a2b5e7730c01/miio/gateway.py#L26

I actually did not program this and it was already in the code like this.
This will not be used for HomeAssistant since there you just set RGB values and HomeAssistant has its own color names defined.
I don't think these functions will be used much.
basically you can now specify "red", "green", "blue", "white", "yellow", "orange", "aqua", "olive" or "purple"
I will put it in the docstrings

Reply:
Ah, okay. Do not add them to the docstring, but add a comment saying that `color_map` variable in the source holds the valid values. This avoids repeating the list :-)

Reply:
Alright, I changed it